operate/client/src/modules/components/ElementInstanceIcon/index.tsx

```diff
@@ -0,0 +1,358 @@
+import FlowNodeEventErrorEnd from 'modules/components/Icon/flow-node-event-error-end.svg?react';
+
+import FlowNodeEventSubprocess from 'modules/components/Icon/flow-node-subprocess-event.svg?react';
+
+import FlowNodeEventTerminateEnd from 'modules/components/Icon/flow-node-event-terminate-end.svg?react';
+import FlowNodeEscalationEndEvent from 'modules/components/Icon/flow-node-escalation-end-event.svg?react';
+import FlowNodeEscalationBoundaryEvent from 'modules/components/Icon/flow-node-escalation-boundary-event.svg?react';
+import FlowNodeEscalationBoundaryNonInterruptingEvent from 'modules/components/Icon/flow-node-escalation-boundary-non-interrupting-event.svg?react';
+import FlowNodeEscalationIntermediateThrowEvent from 'modules/components/Icon/flow-node-escalation-intermediate-throw-event.svg?react';
+import FlowNodeEscalationNonInterruptingStartEvent from 'modules/components/Icon/flow-node-escalation-non-interrupting-start-event.svg?react';
+import FlowNodeEscalationStartEvent from 'modules/components/Icon/flow-node-escalation-start-event.svg?react';
+
+import FlowNodeLinkEventIntermediateCatch from 'modules/components/Icon/flow-node-link-event-intermediate-catch.svg?react';
+import FlowNodeLinkEventIntermediateThrow from 'modules/components/Icon/flow-node-link-event-intermediate-throw.svg?react';
+import FlowNodeEventSignalStart from 'modules/components/Icon/flow-node-event-signal-start.svg?react';
+import FlowNodeEventSignalEnd from 'modules/components/Icon/flow-node-event-signal-end.svg?react';
+import FlowNodeEventSignalIntermediateThrow from 'modules/components/Icon/flow-node-event-signal-intermediate-throw.svg?react';
+import FlowNodeEventSignalIntermediateCatch from 'modules/components/Icon/flow-node-event-signal-intermediate-catch.svg?react';
+import FlowNodeEventSignalInterruptingBoundary from 'modules/components/Icon/flow-node-event-signal-interrupting-boundary.svg?react';
+import FlowNodeEventSignalNonInterruptingBoundary from 'modules/components/Icon/flow-node-event-signal-non-interrupting-boundary.svg?react';
+import FlowNodeEventSignalNonInterruptingStart from 'modules/components/Icon/flow-node-event-signal-non-interrupting-start.svg?react';
+
+import FlowNodeEventCompensationStart from 'modules/components/Icon/flow-node-compensation-start-event.svg?react';
+import FlowNodeEventCompensationEnd from 'modules/components/Icon/flow-node-compensation-end-event.svg?react';
+import FlowNodeEventCompensationIntermediateThrow from 'modules/components/Icon/flow-node-compensation-intermediate-event-throw.svg?react';
+import FlowNodeEventCompensationBoundary from 'modules/components/Icon/flow-node-compensation-boundary-event.svg?react';
+import {useMemo} from 'react';
+
+const getSVGComponent = (
+  businessObject: BusinessObject | undefined,
+  elementInstanceType: ElementInstance['type'],
+) => {
+  if (elementInstanceType === 'AD_HOC_SUB_PROCESS_INNER_INSTANCE') {
```

Comment:
💭 Any reason for not using a switch statement for this comparison chain as well? Would keep the theme of the rest of the function.

Reply:
I refactored to use more switches but left some ifs because I think it's cleaner

Reply:
Looks good! I agree. The remaining `if` statements look totally fine and idiomatic.